neuralforecast/models/timellm.py

```diff
@@ -0,0 +1,497 @@
+    `loss`: PyTorch module, instantiated train loss class from [losses collection](https://nixtla.github.io/neuralforecast/losses.pytorch.html).<br>
+    `valid_loss`: PyTorch module=`loss`, instantiated valid loss class from [losses collection](https://nixtla.github.io/neuralforecast/losses.pytorch.html).<br>
+    `learning_rate`: float=1e-3, Learning rate between (0, 1).<br>
+    `max_steps`: int=1000, maximum number of training steps.<br>
+    `num_lr_decays`: int=-1, Number of learning rate decays, evenly distributed across max_steps.<br>
+    `early_stop_patience_steps`: int=-1, Number of validation iterations before early stopping.<br>
+    `val_check_steps`: int=100, Number of training steps between every validation loss check.<br>
+    `batch_size`: int=32, number of different series in each batch.<br>
+    `valid_batch_size`: int=None, number of different series in each validation and test batch, if None uses batch_size.<br>
+    `windows_batch_size`: int=1024, number of windows to sample in each training batch, default uses all.<br>
+    `inference_windows_batch_size`: int=1024, number of windows to sample in each inference batch.<br>
+    `start_padding_enabled`: bool=False, if True, the model will pad the time series with zeros at the beginning, by input size.<br>
+    `step_size`: int=1, step size between each window of temporal data.<br>
+    `scaler_type`: str='identity', type of scaler for temporal inputs normalization see [temporal scalers](https://nixtla.github.io/neuralforecast/common.scalers.html).<br>
+    `random_seed`: int, random_seed for pytorch initializer and numpy generators.<br>
+    `num_workers_loader`: int=os.cpu_count(), workers to be used by `TimeSeriesDataLoader`.<br>
+    `drop_last_loader`: bool=False, if True `TimeSeriesDataLoader` drops last non-full batch.<br>
+    `alias`: str, optional,  Custom name of the model.<br>
+    `**trainer_kwargs`: int,  keyword trainer arguments inherited from [PyTorch Lighning's trainer](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.trainer.trainer.Trainer.html?highlight=trainer).<br>
+
+    **References:**<br>
+    -[Ming Jin, Shiyu Wang, Lintao Ma, Zhixuan Chu, James Y. Zhang, Xiaoming Shi, Pin-Yu Chen, Yuxuan Liang, Yuan-Fang Li, Shirui Pan, Qingsong Wen. "Time-LLM: Time Series Forecasting by Reprogramming Large Language Models"](https://arxiv.org/abs/2310.01728)
+
+    """
+
+    SAMPLING_TYPE = "windows"
+
+    def __init__(
+        self,
+        h,
+        input_size,
+        patch_len: int = 16,
+        stride: int = 8,
+        d_ff: int = 128,
+        top_k: int = 5,
+        d_llm: int = 768,
+        d_model: int = 32,
+        n_heads: int = 8,
+        enc_in: int = 7,
+        dec_in: int = 7,
+        llm=None,
+        llm_config=None,
+        llm_tokenizer=None,
+        llm_num_hidden_layers=32,
+        llm_output_attention: bool = True,
+        llm_output_hidden_states: bool = True,
+        prompt_prefix: str = None,
```

Comment:
adding `propmpt_prefix: Optional[str] = None` should fix the [linter](https://github.com/Nixtla/neuralforecast/actions/runs/8142491133/job/22252164542?pr=908) errors.